src/lib/core/WeaveFabricState.cpp

```diff
@@ -130,6 +134,22 @@ void WeaveSessionKey::Clear(void)
     ClearSecretData((uint8_t *)&MsgEncKey.EncKey, sizeof(MsgEncKey.EncKey));
 }
 
+void WeaveSessionKey::ComputeNextResumptionMsgIds(void)
+{
+     // When calculating the resumption message ids, it needs to be ensured that the next resumption message id is always ahead of the current message ids.
+     while (ResumptionSendMsgId < NextMsgId.GetValue())
+     {
+         ResumptionSendMsgId = (ResumptionSendMsgId == 0) ? InitialSendMsgId + WEAVE_CONFIG_TUNNEL_SESSION_RESUMPTION_MSG_ID_OFFSET :
```

Comment:
As we discussed offline I propose to change this calculation mechanism.

Reply:
Cloud also has the same logic currently. But they can make an adjustment to allow a range from the resumption msgId within a certain limit. That should cover the case of the device moving ahead of the resumptionmsgIds by a few counts. This would not require any additional changes on device side. However, in practice, I am skeptical this would happen as the device ought to tear down the connection entirely and retry, if the TunnelOpen message fails to go or times out.